### PR TITLE
Thymeleaf form-login template

### LINKED ISSUE #104: 1. Create a Thymeleaf login page
Create a Thymeleaf login page with Spring Security

### PR DESCRIPTION
**Pull request by FelixJacobsen**
Fix #104 

I've used this guide as a guideline for the login template: https://www.youtube.com/watch?v=RhIXP5tjHkk&t=1384s&ab_channel=CodeJava

Everything included within the < head >  < /head > tags is used for design / appearance. I've used Bootstrap, which is a very easy to use CSS framework.  More reading about bootstrap: https://getbootstrap.com/docs/5.1/getting-started/introduction/

Everything included in the within the <body> -> </body> tags is used to display information. Every row that includes "th" is linked to Thymeleaf and is not standardized html syntax. More reading about Thymeleaf: https://www.thymeleaf.org/doc/tutorials/2.1/usingthymeleaf.html

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ have different degree of access throughout the application.
 
 ## 1. E/R diagram
 
-![ER Diagram](src/main/resources/static/image/ERdiagram.png)
+![ER Diagram](src/main/resources/static/images/ERdiagram.png)
```
Comment (LordRekishi) [inline README.md:13]: This should perhaps be its own issue?

**Review by LordRekishi — COMMENTED** [review]
Maybe the logo image also could had been its own issue? And it was quite hard to understand which pull request was the first one and in what order they should be implemented? Maybe add a number in the title of the pulls or something?

File: src/main/java/se/iths/crimedatabase/view/LoginPage.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+@Controller
+public class LoginPage {
+
+    @GetMapping("/login")
+    public String showLoginPage() {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        if (authentication == null || authentication instanceof AnonymousAuthenticationToken)
+            return "login";
+
+        return "redirect:/";
```
Comment (HelenaHalldinITHS) [inline src/main/java/se/iths/crimedatabase/view/LoginPage.java:18]: what is the puropse of this? I thaught this was the default behavior?

Comment (HelenaHalldinITHS) [inline src/main/java/se/iths/crimedatabase/view/LoginPage.java:18]: We can move the error handling to the html document instead and just return "login" if we want

File: src/main/resources/templates/login.html [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+                    <input type="username" name="username" class="form-control" placeholder="Enter username" required autofocus/>
+                </p>
+
+                <p>
+                     <input type="password" name="password" class="form-control" placeholder="Enter password" required />
+                </p>
+
+                <p>
+                    <input type="submit" value="Login" class="btn btn-primary"/>
+                </p>
```
Comment (helenahalldiniths) [inline src/main/resources/templates/login.html:32]: Here is an example of the form part (that has error handling here insted of in our controllers):

```
<form name="f" th:action="@{/signin}" method="post">
                    <fieldset>
                        <!-- Thymeleaf + Spring Security error display -->
                        <div th:if="${param.error}" class="alert alert-danger">
                            Invalid username and password.
                        </div>

                        <div th:if="${param.logout}" class="alert alert-success">
                            You have been logged out.
                        </div>

                        <!-- Login Controls -->
                        <div class="form-group">
                            <label for="txtUsername">Username</label>
                            <input type="text" class="form-control" id="txtUsername" name="txtUsername"
                                   placeholder="Username">
                        </div>

                        <div class="form-group">
                            <label for="txtPassword">Password</label>
                            <input type="password" class="form-control" id="txtPassword" name="txtPassword"
                                   placeholder="Password">
                        </div>

                        <div class="form-check">
                            <input type="checkbox" class="form-check-input" id="checkRememberMe" name="checkRememberMe">
                            <label class="form-check-label" for="checkRememberMe">Remember me?</label>
                        </div>

                        <!-- Login Button -->
                        <div class="form-actions" style="margin-top: 12px;">
                            <button type="submit" class="btn btn-success">Log in</button>
                        </div>
                    </fieldset>
                </form>
```

Comment (helenahalldiniths) [reply src/main/resources/templates/login.html:32]: Note: this form doesn't use the conventions (like /login, username, password etc) we can either change it so it uses the convention or we can modify the configuration i SecurityConfig

**Pull request closed without merging**